cluster-provision/images/vm-image-builder/customize-image.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+set -exuo pipefail
+
+function cleanup() {
+  echo "[customize-image] cleanup"
+  rm -rf "$CLOUD_INIT_ISO"
+  virsh destroy "$DOMAIN_NAME" || true
+  virsh undefine "$DOMAIN_NAME" || true
+
+  if [ $? -ne 0 ]; then
+    rm -f "$NEW_IMAGE"
+  fi
```

Comment:
This is a common bash pattern:
```
[ $? -ne 0 ] && rm -f "$NEW_IMAGE"
```

But this will always run because you do `|| true`

Reply:
so should I move it to line 6?

Reply:
Just run it without any condition, the `-f` will always cause it to succeed.

Reply:
I want to delete  NEW_IMAGE only if there was an error during the script invocation
this is why I added the condition in the first place, it should be moved to line 6
so `|| true` will not interfere with `$?`